### PR TITLE
Various enhancements

### PR DESCRIPTION
**Pull request by jschueller**
fixes:
- #1403
- #1405
- #1416
- #1390
- #1419
- #1404

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -41,6 +41,7 @@
  * #1403 (Python import otagrum throws an error)
  * #1405 (Most of the simulation algorithms for rare event fail on a coronavirus example)
  * #1416 (MethodOfMomentsFactory has no setOptimizationBounds method)
+ * #1419 (BoundingVolumeHierarchy segaults/hangs)
```
Comment (sofianehaddad) [inline ChangeLog]: could you also take some of my changes regarding this issue ? (mainly instanciation of BVH class, see e272f93e18aa292e45c5f3998b3e0eb5027df0e6)

Comment (jschueller) [reply ChangeLog]: done

File: python/src/MethodOfMomentsFactory_doc.i.in [hunk 2]
```diff
@@ -58,6 +58,26 @@ solver : :class:`~openturns.OptimizationAlgorithm`
 
 // ---------------------------------------------------------------------
 
+%feature("docstring") OT::MethodOfMomentsFactory::setOptimizationBounds
+"Accessor to the optimization bounds.
+
+Parameters
+----------
+problem : :class:`~openturns.Interval`
```
Comment (sofianehaddad) [inline python/src/MethodOfMomentsFactory_doc.i.in]: s/problem/bounds

File: python/src/MethodOfMomentsFactory_doc.i.in [hunk 3]
```diff
@@ -58,6 +58,26 @@ solver : :class:`~openturns.OptimizationAlgorithm`
...
+    The bounds used for numerical optimization of the likelihood."
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::MethodOfMomentsFactory::getOptimizationBounds
+"Accessor to the optimization bounds.
+
+Returns
+-------
+problem : :class:`~openturns.Interval`
```
Comment (sofianehaddad) [inline python/src/MethodOfMomentsFactory_doc.i.in]: same here

**Pull request merged**